src/LaunchDarkly.Client/LdClient.cs

```diff
@@ -52,142 +54,125 @@ public bool IsOffline()
             return _configuration.Offline;
         }
 
-        public bool Toggle(string key, User user, bool defaultValue = false)
-        {
-            try
-            {
-                var value = Evaluate(key, user, defaultValue);
-                sendFlagRequestEvent(key, user, value, defaultValue);
-                if (value.Type.Equals(JTokenType.Boolean))
-                {
-                    return value.Value<bool>();
-                }
 
-            }
-            catch (Exception ex)
-            {
-                Logger.Error("Unhandled exception in LaunchDarkly client" + ex.Message);
-            }
-            sendFlagRequestEvent(key, user, defaultValue, defaultValue);
-            return defaultValue;
+      [Obsolete("Please use BoolVariation instead.")]
+      public bool Toggle(string key, User user, bool defaultValue = false)
+      {
+          Logger.Warn("Toggle() method is deprecated. Please use BoolVariation() instead");
+          return BoolVariation(key, user, defaultValue);
+      }
+
+      public bool BoolVariation(string key, User user, bool defaultValue = false)
+        {
+          var value = Evaluate(key, user, defaultValue, JTokenType.Boolean);
+          return value.Value<bool>();
         }
 
         public int IntVariation(string key, User user, int defaultValue)
         {
-            try
```

Comment:
these methods now return the default value when the expected type is not the same as the type returned from feature flag evaluation.